app/common/metrics.py

```diff
@@ -4,3 +4,8 @@
 request_latency = Histogram(  # pylint: disable=no-value-for-parameter
     'request_latency_seconds',
     'Latency of HTTP requests in seconds')
+
+build_state_duration_seconds = Histogram(  # pylint: disable=no-value-for-parameter
```

Comment:
I played with the idea of adding a separate metric `api_queued_state` that's just the sum of the amount of time we spend in Queued, Preparing and Prepared times. If we want a separate metric I can do that, but we can always just sum them in prometheus.